### ISSUE TITLE
Improve the validation of image version

### ISSUE THREAD

**Issue by hkarshad** [hashicorp/packer-plugin-azure#187]
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description
I was recently building an azure-arm image using packer into a share image gallery and had given the value for image_version value as 'snapshot'. It rightly failed due to validation stating the follwoing error message
```
==> azure-arm.ofs-base-image: compute.GalleryImageVersionsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="The image version name is invalid. The image version name should follow Major(int).Minor(int).Patch(int) format, where int is between 0 and 2,147,483,647 (inclusive). For e.g: 1.0.0, 2018.12.1 etc." Target="galleryImageVersionName"
```
While this behaviour is expected but the error appeared after the full process of provisining vm and running all the scripts and then while trying to apture an image. I think this is a bit too late considering the amount of time it takes to get to this error. I feel this should be validated in the early stages of `build`process or may be even as part of `validate` command
A written overview of the feature.
Introduce validation of image version as part of `validate` command or early stages `build` command
#### Use Case(s)

Any relevant use-cases that you see.
Any mistake in image version
#### Potential configuration

```
```

**Comment by surajssd** [comment]
I hit this today, yeah this validation should happen upfront not after the build is finished.

**Comment by JenGoldstrich** [comment]
This will be fixed with the release of v2.0.0 code, for which has been done in this PR https://github.com/hashicorp/packer-plugin-azure/pull/326
